src/sdk/namespace_fs.js

```diff
@@ -2990,6 +3020,65 @@ class NamespaceFS {
         return false;
     }
 
+    /**
+     * @param {nb.NativeFSContext} fs_context
+     * @param {number} [size_hint]
+     * @returns {Promise<void>}
+     */
+    async _throw_if_low_space(fs_context, size_hint) {
+        if (!config.NSFS_SPC_CHECK_ENABLED) return;
+
+        const { statfs } = await nsfs_bucket_statfs_cache.get_with_cache({ bucket_path: this.bucket_path, fs_context });
+        const free_space = (statfs.bfree * statfs.bsize) - (size_hint || 0);
+        const total_space = statfs.blocks * statfs.bsize;
```

Comment:
I worry that these numbers might not fit into js number for large systems.
js number has up to 53 bits for integer values which can fit up to 8 PB (Number.MAX_SAFE_INTEGER), but practically doing calculations with such large numbers is risky and can produce unexpected results.
We could either do all the calculations with BigInt or just convert all units to MB.